src/VirtualClient/VirtualClient.Monitors/Lspci/LspciMonitor.cs

```diff
@@ -71,7 +71,8 @@ private async Task ListPciAsync(EventContext telemetryContext, CancellationToken
             await Task.Delay(this.MonitorWarmupPeriod, cancellationToken);
 
             bool firstRun = true;
-            while (!cancellationToken.IsCancellationRequested)
+            int iteration = 0;
+            while (!cancellationToken.IsCancellationRequested && iteration < this.MonitorIterations)
```

Comment:
The default of this.MonitorIterations is null, I think this will nullref

Reply:
Thanks, I had set default to -1 for infinite loop